### PR TITLE
Add post announcing pulumi-kubernetes-operator

### PR DESCRIPTION
**Pull request by metral**
### Proposed changes

Closes https://github.com/pulumi/docs/issues/3886

### CODE REVIEW

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+date: "2020-08-12"
+title: "Introducing the Pulumi Kubernetes Operator"
+authors: ["mike-metral"]
+tags: ["Kubernetes", "Operator", "Continuous Delivery"]
+meta_desc: "Introducing the Pulumi Kubernetes Operator: Deploy infrastructure in Pulumi Stacks"
+meta_image: operator.png
+---
+
+Kubernetes developers and operators work together to manage workloads
+and to continuously ship software through CI/CD. These users
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:11]:
```suggestion
and continuously ship software through CI/CD. Users
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+have an affinity for automation and pipelines, and richer integration
+with Kubernetes is a growing theme across the cloud native ecosystem.
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:13]:
```suggestion
with Kubernetes is a growing theme across the cloud-native ecosystem.
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+title: "Introducing the Pulumi Kubernetes Operator"
+authors: ["mike-metral"]
+tags: ["Kubernetes", "Operator", "Continuous Delivery"]
+meta_desc: "Introducing the Pulumi Kubernetes Operator: Deploy infrastructure in Pulumi Stacks"
+meta_image: operator.png
+---
+
+Kubernetes developers and operators work together to manage workloads
+and to continuously ship software through CI/CD. These users
+have an affinity for automation and pipelines, and richer integration
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:12]:
```suggestion
with an affinity for automation and pipelines and richer integration
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+with Kubernetes is a growing theme across the cloud native ecosystem.
+
+We're excited to introduce the Pulumi Kubernetes Operator: a Kubernetes
+controller that deploys cloud infrastructure in Pulumi Stacks for you and your
+team.
+
+These program stacks include virtual machines, block storage,
+managed Kuberentes clusters, API resources, serverless functions and more!
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:20]:
```suggestion
managed Kubernetes clusters, API resources, serverless functions, and more!
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+
+<!--more-->
+
+## Overview
+
+The [Pulumi Kubernetes Operator][pulumi-k8s-op] is an [extension pattern][k8s-ext-pattern] that
+enables Kuberentes users to create a Pulumi [`Stack`][stack] as a first-class API
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:27]:
```suggestion
enables Kubernetes users to create a Pulumi [`Stack`][stack] as a first-class API
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+resource, and use the `StackController` to drive the updates of the Stack until
+success.
+
+Check out [how to deploy the operator][deploy-op] to a Kubernetes cluster using any of Pulumi's supported
+languages, or through YAML manifests and `kubectl`.
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:32]:
```suggestion
languages or YAML manifests and `kubectl`.
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 7]
```diff
@@ -0,0 +1,529 @@
...
+
+{{% /choosable %}}
+{{% /chooser %}}
+
+## CI/CD Scenarios
+
+As you build out your Kubernetes workloads, you'll need to implement update strategies
+to ship new software easily and reliably. Kubernetes API resources like
+[Deployments][k8s-deployment] are used to model stateless apps and operate their lifecycles
+e.g. recreate app replicas, or perform a rolling update.
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:431]:
```suggestion
e.g., recreate app replicas, or perform a rolling update.
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 8]
```diff
@@ -0,0 +1,529 @@
...
+
+A common approach that most teams rely on is the [Blue / Green][blue-green] deployment.
+This starts with an initial version of the app, and then deploys an updated version
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:434]:
```suggestion
This starts with an initial version of the app and then deploys an updated version
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 9]
```diff
@@ -0,0 +1,529 @@
...
+alongside the initial version. Once ready, traffic is switched over from the
+initial version to the new version with no downtime.
+
+![Blue Green Kubernetes Deployment](./blue-green.png)
+
+In Kubernetes, Blue/Green is modeled by using a [Service][k8s-service]
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:440]:
```suggestion
In Kubernetes, Blue/Green is modeled using a [Service][k8s-service]
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 10]
```diff
@@ -0,0 +1,529 @@
...
+that load balances and selects a set of Deployment pods labeled "blue," and then switching to a
+different selection of updated, "green" pods.
+
+Similar to the Stack CustomResources in the previous sections, we can deploy a Stack
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:444]:
```suggestion
Like the Stack CustomResources in the previous sections, we can deploy a Stack
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 11]
```diff
@@ -0,0 +1,529 @@
...
+of a [Blue/Green Kubernetes app in Pulumi][blue-green-example], and step through a sequence of its Git commits
+similiar to how a CI/CD pipeline does. Additionally, we can use the [Pulumi config system][pulumi-config] to parameterize
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:446]:
```suggestion
similar to how a CI/CD pipeline does. Additionally, we can use the [Pulumi config system][pulumi-config] to parameterize
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 12]
```diff
@@ -0,0 +1,529 @@
...
+settings and secrets on how to manage the Stack CustomResource project.
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:447]:
```suggestion
settings and secrets to manage the Stack CustomResource project.
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 13]
```diff
@@ -0,0 +1,529 @@
...
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+...
+```
+
+When we're ready to switch traffic, the blue/green
+transition is done by updating the config to the Git new commit of the selector
+swap, and running an update.
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:474]:
```suggestion
swap and running an update.
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 14]
```diff
@@ -0,0 +1,529 @@
...
+
+```bash
+pulumi config set stackCommit f4bf2f7b54ec441d5af374933cca4ef09f2bce24
+pulumi up -y
+```
+
+After a few seconds you should start seeing the generated traffic returning the
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:481]:
```suggestion
After a few seconds, you should start seeing the generated traffic returning the
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 15]
```diff
@@ -0,0 +1,529 @@
...
+{"id":1,"content":"current"}
+{"id":1,"content":"current"}
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+...
+```
+
+Check out the [full walkthrough][blue-green-walkthrough] for a tutorial, and
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:496]:
```suggestion
Check out the [full walkthrough][blue-green-walkthrough] for a tutorial and
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 16]
```diff
@@ -0,0 +1,529 @@
...
+[blue-green-example]: https://github.com/metral/pulumi-blue-green/blob/master/index.ts
+[k8s-deployment]: https://kubernetes.io/docs/concepts/workloads/controllers/deployment
+[blue-green]: https://en.wikipedia.org/wiki/Blue-green_deployment
+[k8s-service]: https://kubernetes.io/docs/concepts/services-networking/service
+[blue-green-walkthrough]: https://github.com/pulumi/pulumi-kubernetes-operator/tree/master/examples/blue-green
+
+## Wrap-Up
+
+The [Pulumi Kubernetes Operator][pulumi-k8s-op] helps deploy Pulumi Stacks in
+your Kubernetes cluster, and naturally fits in with your CI/CD process.
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:510]:
```suggestion
your Kubernetes cluster and naturally fits your CI/CD process.
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 17]
```diff
@@ -0,0 +1,529 @@
...
+
+We covered examples like deploying an [NGINX Stack][pulumi-k8s-nginx] in the same cluster as the
+Operator, creating a new Kubernetes cluster in an [AWS EKS Stack][pulumi-aws-eks], and how Stacks can be
+abstracted for deployment strategies like Blue/Green.
+
+There are [many more][p-examples] examples of deploying infrastructure,
+containers, serverless, and Kubernetes across cloud providers and cloud native services.
```
Comment (spara) [inline content/blog/pulumi-kubernetes-operator/index.md:517]:
```suggestion
containers, serverless, and Kubernetes across cloud providers and cloud-native services.
```

File: content/blog/pulumi-kubernetes-operator/index.md [hunk 18]
```diff
@@ -0,0 +1,529 @@
...
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+{"id":2,"content":"new"}
+...
+```
+
+Check out the [full walkthrough][blue-green-walkthrough] for a tutorial, and
+the video clip below for a demo.
+
+TODO: add video-clip blue/green stack demo.
```
Comment (ericrudder) [inline content/blog/pulumi-kubernetes-operator/index.md:499]: Is there a link for this now?

Comment (metral) [reply content/blog/pulumi-kubernetes-operator/index.md]: The video is in [Gdrive](https://drive.google.com/drive/u/1/folders/156HJjYlQudFOdrPRrVHokJhPlZnoiRFU). Will be uploaded to YT soon

**Pull request merged**